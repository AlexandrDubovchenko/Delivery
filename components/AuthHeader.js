import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { ExitButton } from './ExitButton';
import { TouchableOpacity } from 'react-native-gesture-handler';

export const AuthHeader = (props) => {
    const switchText = props.switch.substring(0, 4) + ' ' + props.switch.substring(4)
    return (
        <View style={styles.header}>
            <ExitButton navigation={props.navigation} />
            <TouchableOpacity onPress={() => props.navigation.replace(props.switch)}><Text style={[styles.button]}>{switchText}</Text></TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    button: {
        color: "#006400",
        marginTop: 25,
        fontSize: 20
    },
    header: {
        flex: .5,
        flexDirection: "row",
        justifyContent: "space-between"
    },
})