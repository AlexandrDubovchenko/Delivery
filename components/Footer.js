import React from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { StyleSheet, View, Image } from 'react-native';
import * as RootNavigation from '../RootNavigation';

const Footer = (props) => {
 
    return (
        <View style={styles.footer}>
            <TouchableOpacity onPress={() => RootNavigation.navigate('HomeScreen')}>
                <Image style={styles.footer_icon} source={require('../assets/Footer/list.png')} />
            </TouchableOpacity>
            <TouchableOpacity onPress={() => RootNavigation.navigate('SearchScreen')}>
                <Image style={styles.footer_icon} source={require('../assets/Footer/search.png')} />
            </TouchableOpacity>
            <TouchableOpacity onPress={() => RootNavigation.navigate('ProfileScreen')}>
                <Image style={styles.footer_icon} source={require('../assets/Footer/person.png')} />
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    footer: {
        flex: 0.1,
        flexDirection: "row",
        justifyContent: "space-around",
        alignItems: "center",
        backgroundColor: '#006400',
    },
    footer_icon: {
        height: 40,
        width: 40
    }
});

export default Footer