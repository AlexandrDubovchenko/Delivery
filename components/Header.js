/* eslint-disable global-require */
import React from 'react';
import { connect } from 'react-redux';
import {
  StyleSheet, View, Text, Image,
} from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import * as RootNavigation from '../RootNavigation';

const styles = StyleSheet.create({
  header: {
    flex: 0.1,
    backgroundColor: '#FFD700',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  buttonImg: {
    height: 40,
    width: 40,
  },
  basketButton: {
    flexDirection: 'row',
    alignItems: 'flex-end',
  },

});

// eslint-disable-next-line no-shadow
const Header = ({ isBasket, basketItemsNumber }) => (
  <View style={styles.header}>
    <TouchableOpacity onPress={() => RootNavigation.goBack()}>
      <Image style={styles.buttonImg} source={require('../assets/Header/back.png')} />
    </TouchableOpacity>

    {!isBasket
      ? (
        <TouchableOpacity onPress={() => RootNavigation.navigate('BasketScreen')}>
          <Image style={styles.buttonImg} source={require('../assets/Header/basket.png')} />
          {
            basketItemsNumber
              ? <Text>{basketItemsNumber}</Text> : null
          }
        </TouchableOpacity>
      )
      : (
        <TouchableOpacity style={styles.basketButton} onPress={() => RootNavigation.navigate('OrderScreen')}>
          <Text>Оформить заказ</Text>
        </TouchableOpacity>
      )}
  </View>
);

const mapStateToProps = (state) => ({
  basketItemsNumber: state.basket.basketList.length,
});

export default connect(mapStateToProps, null)(Header);
