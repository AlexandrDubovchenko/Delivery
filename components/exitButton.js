import React from 'react';
import { Image, StyleSheet } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';

const styles = StyleSheet.create({
  img: {
    marginTop: 20,
    height: 50,
    width: 50,
  },
});

export const ExitButton = (props) => (
  <TouchableOpacity onPress={() => props.navigation.navigate('WelcomeScreen')}>
    <Image
      style={styles.img}
      source={{ uri: 'https://cdn.iconscout.com/icon/free/png-256/close-150-458082.png' }}
    />
  </TouchableOpacity>
);

export default ExitButton;
