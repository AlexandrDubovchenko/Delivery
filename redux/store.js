import { combineReducers, createStore, applyMiddleware } from "redux";
import authReducer from "./reducers/auth-reducer";
import { reducer as formReducer } from 'redux-form'
import thunkMiddleware from "redux-thunk";

const reducers = combineReducers({
    auth: authReducer,
    form: formReducer
})

const store = createStore(reducers, applyMiddleware(thunkMiddleware));

export default store