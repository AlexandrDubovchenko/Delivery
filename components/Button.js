import React from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { StyleSheet } from 'react-native';
import { Text } from '@ui-kitten/components';

const styles = StyleSheet.create({
  orderButton: {
    alignItems: 'center',
    backgroundColor: '#FFD700',
    marginHorizontal: 15,
  },
});

const OrderButton = ({ onSubmit, text }) => (
  <TouchableOpacity
    style={styles.orderButton}
    onPress={onSubmit}
  >
    <Text category="h5">{text}</Text>
  </TouchableOpacity>
);

export default OrderButton;
