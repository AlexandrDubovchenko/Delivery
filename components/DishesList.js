/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { StyleSheet, View, Image } from 'react-native';
import { Card, List, Text } from '@ui-kitten/components';
import { TouchableOpacity } from 'react-native-gesture-handler';

const styles = StyleSheet.create({
  contentContainer: {
    paddingHorizontal: 8,
    paddingVertical: 4,
  },
  item: {
    marginVertical: 4,

  },
  categoryImage: {
    height: 200,
  },
  orderButton: {
    alignItems: 'center',
    backgroundColor: '#FFD700',
  },
});

const DishesList = ({ data, setBasketItem }) => {
  const renderItemHeader = (headerProps, info) => (
    <View {...headerProps}>
      <Text category="h6">
        {info.item.name}
      </Text>
    </View>
  );

  const renderItemFooter = (footerProps, info) => (
    <TouchableOpacity
      style={styles.orderButton}
      onPress={() => setBasketItem(info.item.name, info.item.img)}
    >
      <Text category="h6">
        Заказать
      </Text>
    </TouchableOpacity>
  );


  const renderItem = (info) => (
    <Card
      style={styles.item}
      header={(headerProps) => renderItemHeader(headerProps, info)}
      footer={(footerProps) => renderItemFooter(footerProps, info)}
    >
      <Image style={styles.categoryImage} source={{ uri: info.item.img }} />
    </Card>
  );

  return (
    <List
      style={styles.container}
      contentContainerStyle={styles.contentContainer}
      data={data}
      renderItem={renderItem}
    />
  );
};

export default DishesList;
