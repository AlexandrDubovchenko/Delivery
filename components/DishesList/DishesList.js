import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { getDishesData, resetCategoryListData } from '../../redux/reducers/list-reducer';
import { setBasketItem } from '../../redux/reducers/basket-reducer';
import DishesList from '../DishesList';


const CategoryDishesList = ({
  // eslint-disable-next-line no-shadow
  listItems, title, resetCategoryListData, getDishesData, setBasketItem,
}) => {
  useEffect(() => {
    getDishesData(title.toLowerCase());
  }, []);

  useEffect(() => (
    () => {
      resetCategoryListData();
    }
  ), []);

  return <DishesList setBasketItem={setBasketItem} data={listItems} />;
};

const mapStateToProps = (state) => (
  { listItems: state.list.dishes }
);

export default connect(mapStateToProps,
  { getDishesData, resetCategoryListData, setBasketItem })(CategoryDishesList);
