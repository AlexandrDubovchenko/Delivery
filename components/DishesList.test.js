import TestRenderer from 'react-test-renderer';
import * as eva from '@eva-design/eva';
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ApplicationProvider } from '@ui-kitten/components';
import DishesList from './DishesList';

const initialStateValues = {}

it('DishesList renders correctly', () => {
  const store = createStore((state) => state, initialStateValues)
  const tree = TestRenderer.create(
    <ApplicationProvider {...eva} theme={eva.light}>
      <Provider store={store}>
        <DishesList/>
    </Provider>
    </ApplicationProvider>

  ).toJSON()
  
  expect(tree).toMatchSnapshot()
})