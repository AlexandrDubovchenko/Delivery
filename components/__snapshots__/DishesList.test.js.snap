// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DishesList renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    appearance="default"
    contentContainerStyle={
      Object {
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      }
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "#F7F9FC",
        },
        undefined,
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
